server/v2/stf/stf.go

```diff
@@ -80,19 +84,11 @@ func (s STF[T]) deliverTx(ctx context.Context, state store.WritableState, txByte
 	}
 
 	execResp, execGas, execEvents, err := s.execTx(ctx, state, tx.GetGasLimit()-validateGas, tx)
```

Comment:
just note, if we have any logs or similar due to execution error that utilize GasLimit, it won't actually reflect what the user provided, but what we deduce here. Just a UX thing I want to point out.

Reply:
this is true, let's tune this when we are doing the gasKV impl.